ciri/api/api.c

```diff
@@ -78,8 +78,9 @@ static iota_api_command_t get_command(char const *const command) {
  * Public functions
  */
 
-retcode_t iota_api_get_node_info(iota_api_t const *const api, get_node_info_res_t *const res) {
-  if (api == NULL || res == NULL) {
+retcode_t iota_api_get_node_info(iota_api_t const *const api, get_node_info_res_t *const res,
+                                 error_res_t **const error) {
```

Comment:
why double pointer?

Reply:
Because the response will be allocated only when an error occurs

Reply:
if so, you should remove the const qualifier. And maybe make error_res_t as a private inside the API methods, so we don't need to pass it. We check retcode_t if there is an error.

Reply:
The `const` qualifier is on the second pointer so it's ok since it won't be changed.
This is on the API level, not on the HTTP API level. So the error needs to be passed as parameter because it will be serialized on the HTTP API level.